Build failure: zenpower (linuxPackages_testing)

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

This failure only happens  on Linux 6.16-rc\* due to https://github.com/torvalds/linux/commit/bcbb65559532891148d990527e9df6b8fc98e98d.

To reproduce:
```nix
boot.kernelPackages = pkgs.linuxPackages_testing;
boot.extraModulePackages = [ config.boot.kernelPackages.zenpower ];
boot.kernelModules = [ "zenpower" ];
```



### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/z5jqa4va78pah7fjsc6rr6zgwklqxwnf-zenpower-unstable-2025-02-28.drv' failed with exit code 2;
       last 23 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/32r2xbq9fljvkjfikn5sk2jxjgln3aql-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/00zrahbb32nzawrmv9sjxn36h7qk9vrs-bash-5.2p37/bin/bash KERNEL_BUILD=/nix/store/z4qlchyqdkhl3qx7bhlbi355pm06kc55-linux-6.16-rc4-dev/lib/modules/6.16.0-rc4/build
       > make[1]: Entering directory '/nix/store/z4qlchyqdkhl3qx7bhlbi355pm06kc55-linux-6.16-rc4-dev/lib/modules/6.16.0-rc4/build'
       > make[2]: Entering directory '/build/source'
       >   CC [M]  zenpower.o
       > zenpower.c:39:10: fatal error: asm/amd_nb.h: No such file or directory
       >    39 | #include <asm/amd_nb.h>
       >       |          ^~~~~~~~~~~~~~
       > compilation terminated.
       > make[4]: *** [/nix/store/z4qlchyqdkhl3qx7bhlbi355pm06kc55-linux-6.16-rc4-dev/lib/modules/6.16.0-rc4/source/scripts/Makefile.build:287: zenpower.o] Error 1
       > make[3]: *** [/nix/store/z4qlchyqdkhl3qx7bhlbi355pm06kc55-linux-6.16-rc4-dev/lib/modules/6.16.0-rc4/source/Makefile:2003: .] Error 2
       > make[2]: *** [/nix/store/z4qlchyqdkhl3qx7bhlbi355pm06kc55-linux-6.16-rc4-dev/lib/modules/6.16.0-rc4/source/Makefile:248: __sub-make] Error 2
       > make[2]: Leaving directory '/build/source'
       > make[1]: *** [/nix/store/z4qlchyqdkhl3qx7bhlbi355pm06kc55-linux-6.16-rc4-dev/lib/modules/6.16.0-rc4/source/Makefile:248: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/z4qlchyqdkhl3qx7bhlbi355pm06kc55-linux-6.16-rc4-dev/lib/modules/6.16.0-rc4/build'
       > make: *** [Makefile:29: modules] Error 2
       For full logs, run:
         nix log /nix/store/z5jqa4va78pah7fjsc6rr6zgwklqxwnf-zenpower-unstable-2025-02-28.drv
error: 1 dependencies of derivation '/nix/store/r4wrh9kqjjyc5pl7kgg9q2wcrjvzmnhh-linux-6.16-rc4-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xmrpwfmg6yrsq5dqlmx3k8v3xiy9y5kj-nixos-system-ARISU-25.11.20250707.73feb5e.drv' failed to build
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.0-rc4, NixOS, 24.11 (Vicuna), 24.11.20241024.8b052aa`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixos, nixos-hardware"`
 - nixpkgs: `/nix/store/iyp0bwybs2a0p72prrvl7d3jdmh3mjgh-source`



### Notify maintainers

@alexbakker @Artturin

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/zenpower/default.nix
+++ b/pkgs/os-specific/linux/zenpower/default.nix
@@ -2,18 +2,18 @@
   lib,
   stdenv,
   kernel,
-  fetchFromGitLab,
+  fetchFromGitHub,
 }:
 
 stdenv.mkDerivation rec {
   pname = "zenpower";
-  version = "unstable-2025-02-28";
+  version = "unstable-2025-06-17";
 
-  src = fetchFromGitLab {
-    owner = "shdwchn10";
+  src = fetchFromGitHub {
+    owner = "AliEmreSenel";
     repo = "zenpower3";
-    rev = "138fa0637b46a0b0a087f2ba4e9146d2f9ba2475";
-    sha256 = "sha256-kLtkG97Lje+Fd5FoYf+UlSaEyxFaETtXrSjYzFnHkjY=";
+    rev = "41e042935ee9840c0b9dd55d61b6ddd58bc4fde6";
+    hash = "sha256-0U/JmEd6OJJeUm1ZLFYxpKH15n7+QTWYOgtKIFAuf/4=";
   };
 
   hardeningDisable = [ "pic" ];

